Apps/UnitTests/Shared/Tests.h

```diff
@@ -98,6 +98,82 @@ TEST(NativeAPI, LifeCycle)
     }
 }
 */
+
+TEST(Performance, Spheres)
+{
+    // create a bunch of sphere, does the rendering for a number of frames, log time it took
+    static const std::string script{ R"(
+        console.log("Setting up Performance test.");
+        var engine = new BABYLON.NativeEngine();
+        var scene = new BABYLON.Scene(engine);
+
+        var size = 12;
+        for (var i = 0; i < size; i++) {
+            for (var j = 0; j < size; j++) {
+                for (var k = 0; k < size; k++) {
+                    var sphere = BABYLON.Mesh.CreateSphere("sphere" + i + j + k, 32, 0.9, scene);
+                    sphere.position.x = i;
+                    sphere.position.y = j;
+                    sphere.position.z = k;
+                }
+            }
+        }
+
+        scene.createDefaultCamera(true, true, true);
+        scene.activeCamera.alpha += Math.PI;
+        scene.createDefaultLight(true);
+        engine.runRenderLoop(function () {
+            scene.render();
+        });
+        console.log("Ready!");
+        SetReady();
+        )" };
+
+    Babylon::Graphics::Device device = deviceTestConfig;
+    std::optional<Babylon::Graphics::DeviceUpdate> update{};
+    std::promise<int32_t> ready;
+    update.emplace(device.GetUpdate("update"));
+
+    Babylon::AppRuntime runtime{};
+    runtime.Dispatch([&ready, &device](Napi::Env env) {
+        device.AddToJavaScript(env);
+
+        Babylon::Polyfills::Console::Initialize(env, [](const char* message, auto) {
+            printf("%s", message);
+            fflush(stdout);
+            });
+        Babylon::Polyfills::Window::Initialize(env);
+        Babylon::Plugins::NativeEngine::Initialize(env);
+        env.Global().Set("SetReady", Napi::Function::New(env, [&ready](const Napi::CallbackInfo& info)
+        {
+            Napi::Env env = info.Env();
+            ready.set_value(1);
+        }, "SetReady"));
+    });
+
+    Babylon::ScriptLoader loader{ runtime };
+    loader.LoadScript("app:///Scripts/babylon.max.js");
+    loader.LoadScript("app:///Scripts/babylonjs.materials.js");
+    loader.Eval(script, "code");
+
+    ready.get_future().get();
+
+    const auto start = std::chrono::high_resolution_clock::now();
+
+    for (int frame = 0; frame < 100; frame++)
+    {
+        device.StartRenderingCurrentFrame();
+        update->Start();
+        update->Finish();
+        device.FinishRenderingCurrentFrame();
+    }
+    // Stop measuring time
+    const auto stop = std::chrono::high_resolution_clock::now();
+    const auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(stop - start);
+    const float durationSeconds = float(double(duration.count()) / 1000.);
```

Comment:
nit: Why convert to double than back to float? Can you not convert to float directly?

Reply:
I want to keep precision until the /1000.

Reply:
Divide by 1000 shouldn't change the precision much, but I'm okay if you want to keep it.

Reply:
Fixed